gcloud/storage/_helpers.py

```diff
@@ -79,10 +79,12 @@ def _patch_property(self, name, value):
     def _set_properties(self, value):
         """Set the properties for the current object.
 
-        :type value: dict
+        :type value: dict or :class:`gcloud.storage.batch._FutureDict`
         :param value: The properties to be set.
         """
         self._properties = value
+        if hasattr(value, 'owner'):
+            value.owner = self
```

Comment:
Because the setter methods are all going to raise exceptions, `self._changes` is irrelevant "future blob" or "future bucket":  we don't need to call `_set_properties` until we have a *real* dict.  When creating a blob/bucket inside the context of a batch, we can just assign a singleton `_FutureDict` instance to `_properties` and move on.